staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -147,6 +148,10 @@ func (i *indexedWatchers) deleteWatcher(number int, value string, supported bool
 }
 
 func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cacheWatcher)) {
+	// note that we don't have to call setDrainInputBuffer method on the watchers
```

Comment:
food for thought: for the sake of limiting size of this PR, maybe we can move support for `drainingWatchers` (without the logic deciding when to use it it ) to a separate PR and just merge it...

Reply:
I went ahead and did that in: https://github.com/kubernetes/kubernetes/pull/111460